battDB/models.py

```diff
@@ -530,6 +530,8 @@ class ExperimentDataFile(cm.BaseModel):
         related_name="data_files",
     )
 
+    devices = models.ManyToManyField(Device, related_name="data_files")
```

Comment:
This makes the devices field mandatory for the EDF despite it not being mandatory for the Experiment itself (and it's unclear to me if it should be made mandatory in https://github.com/ImperialCollegeLondon/Faraday-liionsden/issues/118#issuecomment-1324969971). Is this ok?

Reply:
Yeah this is the desired behaviour - users are able to create an experiment, then later edit it to  add experimental data, potentially multiple times over many months. So the experiment could be defined without any devices, but each time an EDF is added it stands to reason some device must have been involved in the experiment.